src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -95,8 +95,7 @@ internal sealed class TupleTypeSymbol : WrappedNamedTypeSymbol
             Debug.Assert(elementNames.IsDefault || elementTypes.Length == elementNames.Length);
 
             int numElements = elementTypes.Length;
-
-            if (numElements <= 1)
+            if (numElements == 0)
```

Comment:
Ok, I believe this condition should be relaxed only when this helper is called from CommonCreateTupleTypeSymbol. So, either we should create a specialized helper for CommonCreateTupleTypeSymbol to use and share as much code as possible across both of them, or we could add an optional parameter that would explicitly allow the relaxation (the relaxation should be disallowed by default).